dali/operators/image/crop/bbox_crop.cc

```diff
@@ -14,143 +14,635 @@
+    while (!prospective_crop.success) {
+      prospective_crop = FindProspectiveCrop(bounding_boxes, labels, sample);
+    }
+
+    WriteCropToOutput(ws, prospective_crop.crop);
+
+    const auto &selected_boxes  = prospective_crop.boxes;
+    WriteBoxesToOutput(ws, selected_boxes);
+
+    if (has_labels_) {
+      const auto &selected_labels = prospective_crop.labels;
+      DALI_ENFORCE(selected_boxes.size() == selected_labels.size(),
+        make_string("Expected boxes.size() == labels.size(). Received: ", selected_boxes.size(),
+          " != ", selected_labels.size()));
+      WriteLabelsToOutput(ws, selected_labels);
+    }
+  }
+
+ private:
+  struct ProspectiveCrop {
+    bool success = false;
+    Box<ndim, float> crop{};
+    std::vector<Box<ndim, float>> boxes;
+    std::vector<int> labels;
+
+    ProspectiveCrop(bool success,
+                    const Box<ndim, float> &crop,
+                    const std::vector<Box<ndim, float>> &boxes,
+                    const std::vector<int> &labels)
+        : success(success), crop(crop), boxes(boxes), labels(labels) {}
+    ProspectiveCrop() = default;
+  };
+
+  const ProspectiveCrop FindProspectiveCrop(const std::vector<Box<ndim, float>> &bounding_boxes,
+                                            const std::vector<int> &labels,
+                                            int sample) {
+    auto &rng = rngs_[sample];
+    std::uniform_int_distribution<> idx_dist(0, sample_options_.size() - 1);
+    SampleOption option = sample_options_[idx_dist(rng)];
+    bool absolute_crop_dims = has_crop_shape_;
+
+    if (option.no_crop) {
+      Box<ndim, float> no_crop = Uniform<ndim>(0.0f, 1.0f);
+      if (absolute_crop_dims) {
+        auto &input_shape = input_shape_[sample];
+        for (int d = 0; d < ndim; d++)
+          no_crop.hi[d] *= input_shape[d];
+      }
+      return ProspectiveCrop(true, no_crop, bounding_boxes, labels);
+    }
+
+    std::array<float, coords_size> shape, anchor, out_bounds, rel_bounds;
+    for (int i = 0; i < num_attempts_; i++) {
+      if (absolute_crop_dims) {
+        auto &crop_shape = crop_shape_[sample];
+        auto &input_shape = input_shape_[sample];
+
+        for (int d = 0; d < ndim; d++) {
+          shape[d] = static_cast<float>(crop_shape[d]);
+          out_bounds[ndim + d] = shape[d];
+          rel_bounds[ndim + d] = shape[d] / input_shape[d];
+        }
+
+        for (int d = 0; d < ndim; d++) {
+          std::uniform_int_distribution<> anchor_dist(0, input_shape[d] - crop_shape[d]);
+          anchor[d] = static_cast<float>(anchor_dist(rng));
+          out_bounds[d] = anchor[d];
+          rel_bounds[d] = anchor[d] / input_shape[d];
+        }
+      } else {  // relative dimensions
+        std::uniform_real_distribution<float> extent_dist(scale_range_.min, scale_range_.max);
+        for (int d = 0; d < ndim; d++) {
+          shape[d] = extent_dist(rng);
+          out_bounds[ndim + d] = rel_bounds[ndim + d] = shape[d];
```

Comment:
```suggestion
          out_bounds[ndim + d] = shape[d];
          rel_bounds[ndim + d] = shape[d];
```
no strong feelings about it.